utilities/xmlrpcpp/src/XmlRpcValue.cpp

```diff
@@ -610,7 +611,28 @@ namespace XmlRpc {
       default:           break;
       case TypeBoolean:  os << _value.asBool; break;
       case TypeInt:      os << _value.asInt; break;
-      case TypeDouble:   os << _value.asDouble; break;
+      case TypeDouble:
+        {
+          static std::once_flag once;
+          char buf[128]; // Should be long enough
+          int required_size = std::snprintf(buf, sizeof(buf)-1,
+                                getDoubleFormat().c_str(), _value.asDouble);
+          if (required_size < 0) {
+            std::call_once(once,
+              [](){XmlRpcUtil::error("Failed to format with %s", getDoubleFormat().c_str());});
+            os << _value.asDouble;
+          } else if (required_size >= 0 && required_size < static_cast<int>(sizeof(buf))) {
```

Comment:
The `required_size >= 0` part of the condition is redundant since the the previous `if` evaluated to `false` to enter this `else` block.

Reply:
correct, thanks.